Readability.js

```diff
@@ -232,6 +232,25 @@ Readability.prototype = {
     var prePath = this._uri.prePath;
     var pathBase = this._uri.pathBase;
 
+    // Use `<base>` tag's href if it exists.
+    var bases = this._doc.getElementsByTagName("base");
+    var baseHref = null;
+    if (bases.length) {
+      baseHref = bases[0].getAttribute("href");
+      if (baseHref) {
+        if (baseHref.slice(-1) === "/") {
+          prePath = baseHref.slice(0, -1);
```

Comment:
I think this is wrong. `prePath` is the bit of the URL before the path, including domain but obviously not including the path itself.

The URL in a base tag can be both absolute and relative. If this is an absolute URL with a path, you'll set `prePath` to something including a path. If it's relative and includes a path, you'll still include that path as well.

Generally, doing this resolution in here seems pretty difficult.

Why don't we use document.baseURI when determining the URI to pass to reader mode in the consumer?